remote/remote_state_s3.go

```diff
@@ -549,6 +557,48 @@ func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, te
 	return nil
 }
 
+// Add a policy to enforce TLS based access to the bucket
+func EnableEnforcedTLSAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	terragruntOptions.Logger.Printf("Enabling enforced TLS access for S3 bucket %s", config.Bucket)
+
+	tlsS3Policy := map[string]interface{}{
+		"Version": "2012-10-17",
+		"Statement": []map[string]interface{}{
+			{
+				"Sid":    "AllowSSLRequestsOnly",
+				"Action": "s3:*",
+				"Effect": "Deny",
+				"Resource": []string{
+					"arn:aws:s3:::" + config.Bucket,
+					"arn:aws:s3:::" + config.Bucket + "/*",
+				},
+				"Condition": map[string]interface{}{
+					"Bool": map[string]interface{}{
+						"aws:SecureTransport": "false",
```

Comment:
Should this be true?

Reply:
No. This is correct given that the effect is set to deny if the condition matches - and it will match if non-encrypted transport is being used.